graph/appender/response_time.go

```diff
@@ -48,60 +50,98 @@ func (a ResponseTimeAppender) appendGraph(trafficMap graph.TrafficMap, namespace
 	}
 	log.Warningf("Generating responseTime using quantile [%.2f]", quantile)
 
-	// query prometheus for the responseTime info in two queries. The first query gathers responseTime for
-	// requests originating outside of the namespace...
-	namespacePattern := fmt.Sprintf(".*\\\\.%v\\\\..*", namespace)
-	query := fmt.Sprintf("histogram_quantile(%.2f, sum(rate(%s{source_service!~\"%v\",destination_service=~\"%v\",response_code=\"200\"}[%vs])) by (%s))",
+	// query prometheus for the responseTime info in three queries:
+	// 1) query for responseTime originating from "unknown" (i.e. the internet)
+	groupBy := "le,source_workload_namespace,source_workload,source_app,source_version,destination_service_namespace,destination_service_name,destination_workload,destination_app,destination_version"
```

Comment:
what is "le" ?

Reply:
"le" is a prometheus thing, it's a "bucket label" when working with histogram quantiles. See https://prometheus.io/docs/practices/histograms/#quantiles.